lib/components/side_drawer.dart

```diff
@@ -0,0 +1,87 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:font_awesome_flutter/font_awesome_flutter.dart';
+import 'package:runningapp/pages/logged_in/providers/user_info_provider.dart';
+
+class SideDrawer extends ConsumerWidget {
+  final Function(int) onTap;
+
+  const SideDrawer({required this.onTap, super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final userInfo = ref.watch(userInformationProvider).asData?.value;
+
+    final name = userInfo?['name'] as String?;
+    final points = userInfo?['points'] as int?;
+
+    return Drawer(
+      child: ListView(
+        children: [
+          UserAccountsDrawerHeader(
+            accountName:
+                name != null ? Text(name) : const CircularProgressIndicator(),
+            accountEmail: Text('$points pts'),
+            currentAccountPicture: CircleAvatar(
+              child: ClipOval(
+                child: Image.network(
```

Comment:
Consider using default placeholder image if you are using a Image.network in case the image cant be retrieved.

Eg:
```
new Container(
      child: FadeInImage.assetNetwork(
          placeholder: 'assets/place_holder.jpg',
          image:url
      )
```